custom_components/weenect/sensor.py

```diff
@@ -124,7 +198,23 @@ class WeenectSensor(WeenectEntity, SensorEntity):
     def native_value(self) -> StateType | datetime:
         """Return the state of the resources if it has been received yet."""
         if self.id in self.coordinator.data:
-            return self.coordinator.data[self.id][self.entity_description.key]
+            if self.entity_description.key == "call_available":
+                if (
+                    "call_usage" in self.coordinator.data[self.id]
+                    and "call_max_threshold" in self.coordinator.data[self.id]
+                ):
+                    return (
+                        self.coordinator.data[self.id]["call_max_threshold"]
+                        - self.coordinator.data[self.id]["call_usage"]
+                    )
+                else:
+                    return None
```

Comment:
Please put that in a separate private method. e.g. `get_call_available(self)`

Reply:
You need to help me out here. I'm simply no developer. Having a hard time understanding how to make this a private method within that class.

Reply:
You are 99% there. In Python nothing is really private like in other languages like Java or C#. There is however the convention that methods that start with an underscore `_` should not be used from outside the class. So you only have to name it `def _get_call_available(self)` and you are done. You can also remove the `@property` annotation. It is not needed. It would allow the function to be accessed like a property so `sensor.get_call_available` instead of `sensor.get_call_available()` (note the paranthesis)

Bonus points if you add the return type to the function.

`def _get_call_available(self) -> int:`

Reply:
Aaaah I see the issue with the failing tests. That is because of the `@property` annotation as I have written above.

Reply:
Thank you! That helped a lot and made me learn something new.